common/base/tests/Test_DevicePreallocatedAllocator.cu

```diff
@@ -0,0 +1,280 @@
+/*
+* Copyright 2019-2020 NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "gtest/gtest.h"
+
+#include <vector>
+
+#include <claraparabricks/genomeworks/types.hpp>
+#include <claraparabricks/genomeworks/utils/cudautils.hpp>
+#include <claraparabricks/genomeworks/utils/device_preallocated_allocator.cuh>
+
+namespace claraparabricks
+{
+
+namespace genomeworks
+{
+
```

Comment:
Would it make sense to move the DevicePreallocatedAllocator to a `detail` namespace? We don't want users to use this component directly, right?

Reply:
Yes, done